core/src/main/java/hudson/scm/ChangeLogParser.java

```diff
@@ -42,6 +42,17 @@
 public abstract class ChangeLogParser {
 
     /**
+     * @since
+     */
+    public ChangeLogSet<? extends Entry> parse(Run build, SCM scm, File changelogFile) throws IOException, SAXException {
+        if (build instanceof AbstractBuild && Util.isOverridden(ChangeLogParser.class, getClass(), "parse", AbstractBuild.class, SCM.class, File.class)) {
+            return parse((AbstractBuild) build, scm, changelogFile);
+        } else {
+            throw new AbstractMethodError("You must override the newer overload of parse");
```

Comment:
"... to support non-Pipeline jobs".

Ideally we need a `@RequireOverride` annotation to make these cases explicit during the compilation/static analysis